File: src/java/main/commands/RichJson_folder.java
package commands;

import core.RichJsonCommand;
import core.RichJsonConstants;
import other.RichJsonConfig;
import core.RichJsonContext;
import helper.RichJsonFileHelper;
import core.RichJsonParser;

/**
 * Liest den Datenbaum eines gegebenen Ordnerpfads ein.
 */
public class RichJson_folder implements RichJsonCommand {

    @Override
    public Object execute(RichJsonParser parser, RichJsonContext context) {
        String folderPath = (String) context.currentMember;
        return RichJsonFileHelper.readDirectory(folderPath, true);
    }
}

File: src/c++/main/commands/RichJson_folder.cpp
#include "RichJson_folder.hpp"
#include "../core/RichJsonContext.hpp"
#include "../helper/RichJsonFileHelper.hpp"

namespace RichJson {

json richJson_folder(RichJsonParser&, RichJsonContext& context) {
    std::string folderPath = context.currentMember.get<std::string>();
    return RichJsonFileHelper::readDirectory(folderPath, true);
}

} // namespace RichJson
